mne_bids_pipeline/steps/sensor/_99_group_average.py

```diff
@@ -722,18 +722,20 @@ def average_csp_decoding(
     all_decoding_data_time_freq = []
     for key in list(in_files):
         fname_xlsx = in_files.pop(key)
-        decoding_data_freq = pd.read_excel(
-            fname_xlsx,
-            sheet_name="CSP Frequency",
-            dtype={"subject": str},  # don't drop trailing zeros
-        )
-        decoding_data_time_freq = pd.read_excel(
-            fname_xlsx,
-            sheet_name="CSP Time-Frequency",
-            dtype={"subject": str},  # don't drop trailing zeros
-        )
-        all_decoding_data_freq.append(decoding_data_freq)
-        all_decoding_data_time_freq.append(decoding_data_time_freq)
+        with pd.ExcelFile(fname_xlsx) as xf:
+            decoding_data_freq = pd.read_excel(
+                xf,
+                sheet_name="CSP Frequency",
+                dtype={"subject": str},  # don't drop trailing zeros
+            )
+            all_decoding_data_freq.append(decoding_data_freq)
+            if "CSP Time-Frequency" in xf.sheet_names:
```

Comment:
@hoechenberger found a nicer API than the try/except

Reply:
looks good!